sile's autoconf files are outdated and needs update

Hello, I have encountered the following error message when porting sile to the RISC-V platform, following the build script here[2]:

```
This script, last modified 2014-03-23, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (build-aux/config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.
```

(original log can be seen here[1].)

Please consider updating the Autoconf version.
Thanks :)

[1]: https://archriscv.felixc.at/.status/log.htm?url=logs/sile/sile-0.13.0-1.log
[2]: https://github.com/archlinux/svntogit-community/blob/packages/sile/trunk/PKGBUILD#L59

However, another error occurs after a successful build, when I'm running `make check` on a RISC-V vm:

```
[root@host sile-0.13.0]# make check
output=$(mktemp -t selfcheck-XXXXXX.pdf)
trap 'rm -f $output' EXIT HUP TERM
echo "<sile>foo</sile>" | ./sile -o $output -
/usr/bin/pdfinfo $output | /usr/bin/grep "SILE v0.13.0"
SILE v0.13.0 (Lua 5.4)
</dev/stdin>

Error detected:
        /usr/share/lua/5.4/cassowary/init.lua:488: attempt to concatenate a nil value (field 'value')
Syntax Error: Document stream is empty
make: *** [Makefile:1400: selfcheck] Error 1
```

It's not occurring on an x86 laptop:

```
src/sile-0.13.0 community/packages/sile> make check
touch .built-subdirs
output=$(mktemp -t selfcheck-XXXXXX.pdf)
trap 'rm -f $output' EXIT HUP TERM
echo "<sile>foo</sile>" | ./sile -o $output -
/usr/bin/pdfinfo $output | /usr/bin/grep "SILE v0.13.0"
SILE v0.13.0 (Lua 5.4)
</dev/stdin>

! Error loading localizations en: no localizations for this language at /dev/stdin: in \vfill
[1]
Producer:        SILE v0.13.0 (Lua 5.4)
......
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -31,10 +31,6 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install fonts-sil-gentiumplus libarchive-tools libfontconfig1-dev libharfbuzz-dev libicu-dev liblua5.3-dev libpng-dev lua5.3 lua-sec lua-socket lua-zlib-dev luarocks poppler-utils
-      - name: Monkey patch decrepit system packages
-        run: |
-          curl -fsSL https://github.com/luarocks/luarocks/commit/9ff512e35455939f02eaec2318e3acc77782fdeb.patch |
-            sudo patch -d /usr/share/lua/5.1 -p2
       - name: Configure
         run: |
           ./bootstrap.sh

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,6 +1,6 @@
 name: Build
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -5,7 +5,7 @@ on: [push, pull_request]
 jobs:
 
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/commitlint.yml
+++ b/.github/workflows/commitlint.yml
@@ -1,6 +1,6 @@
 name: Commitlint
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 

--- a/.github/workflows/commitlint.yml
+++ b/.github/workflows/commitlint.yml
@@ -5,7 +5,7 @@ on: [push, pull_request]
 jobs:
 
   commitlint:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -13,7 +13,7 @@ jobs:
   ghcr:
     strategy:
       fail-fast: false
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/luacheck.yml
+++ b/.github/workflows/luacheck.yml
@@ -5,7 +5,7 @@ on: [push, pull_request]
 jobs:
 
   luacheck:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/luacheck.yml
+++ b/.github/workflows/luacheck.yml
@@ -1,6 +1,6 @@
 name: Luacheck
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 

--- a/.github/workflows/nix.yml
+++ b/.github/workflows/nix.yml
@@ -5,7 +5,7 @@ on: [push, pull_request]
 jobs:
 
   nix-flake:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/nix.yml
+++ b/.github/workflows/nix.yml
@@ -1,6 +1,6 @@
 name: Run Flake
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -15,7 +15,7 @@ jobs:
           - [ '5.1', 'CFLAGS="-O2 -Wall -DLUA_USE_LINUX -fPIC"']
           - [ 'luajit', 'XCFLAGS=-fPIC' ]
           - [ 'luajit-openresty', 'XCFLAGS=-fPIC' ]
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     name: Test on Lua ${{ matrix.luaVersion[0] }}
     steps:
       - name: Checkout

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -1,6 +1,6 @@
 name: Test
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 
@@ -12,7 +12,7 @@ jobs:
           - [ '5.4.2', 'MYCFLAGS=-fPIC' ] # https://github.com/luarocks/luarocks/issues/1302
           - [ '5.3', 'MYCFLAGS=-fPIC' ]
           - [ '5.2', 'MYCFLAGS=-fPIC' ]
-          - [ '5.1', 'CFLAGS="-O2 -Wall -DLUA_USE_LINUX -fPIC"']
+          - [ '5.1', 'CFLAGS="-O2 -Wall -DLUA_USE_LINUX -fPIC"' ]
           - [ 'luajit', 'XCFLAGS=-fPIC' ]
           - [ 'luajit-openresty', 'XCFLAGS=-fPIC' ]
     runs-on: ubuntu-22.04

--- a/.github/workflows/versioning.yml
+++ b/.github/workflows/versioning.yml
@@ -7,7 +7,7 @@ on:
 jobs:
 
   actions-tagger:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/.github/workflows/versioning.yml
+++ b/.github/workflows/versioning.yml
@@ -2,7 +2,7 @@ name: Versioning
 
 on:
   release:
-    types: [created, published, edited]
+    types: [ created, published, edited ]
 
 jobs:
 

--- a/README.md
+++ b/README.md
@@ -230,8 +230,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - name: The SILE Typesetter
+        uses: actions/checkout@v3
+      - name: Render document with SILE
         uses: sile-typesetter/sile@v0
         with:
           args: my-document.sil

--- a/README.md
+++ b/README.md
@@ -224,7 +224,7 @@ As a case study, here is how a workflow could be setup in GitHub Actions:
 
 ```yaml
 name: SILE
-on: [push, pull_request]
+on: [ push, pull_request ]
 jobs:
   sile:
     runs-on: ubuntu-latest

--- a/documentation/c02-gettingstarted.sil
+++ b/documentation/c02-gettingstarted.sil
@@ -319,7 +319,7 @@ As a case study, here is how a work flow could be setup in GitHub Actions:
 \begin{verbatim}
 \line
 name: SILE
-on: [push, pull_request]
+on: [ push, pull_request ]
 jobs:
   sile:
     runs-on: ubuntu-20.04

--- a/documentation/c02-gettingstarted.sil
+++ b/documentation/c02-gettingstarted.sil
@@ -322,14 +322,13 @@ name: SILE
 on: [ push, pull_request ]
 jobs:
   sile:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     name: SILE
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - name: SILE
-        id: sile
-        uses: docker://siletypesetter/sile:latest
+        uses: actions/checkout@v3
+      - name: Render document with SILE
+        uses: sile-typesetter/sile@v0
         with:
           args: my-document.sil
 \line

